ui/src/overview.tsx

```diff
@@ -321,27 +399,93 @@ function DemographicCharts({ cohort }: DemographicChartsProps) {
 
   const api = useContext(EntityCountsApiContext);
 
-  const barColours = [
-    "#003f5c",
-    "#2f4b7c",
-    "#665191",
-    "#a05195",
-    "#d45087",
-    "#f95d6a",
-    "#ff7c43",
-    "#ffa600",
-  ];
+  const generatePropertyString = (
+    property: ChartConfigProperty,
+    entityCountStruct: tanagra.EntityCountStruct
+  ) => {
+    let propertyString = "";
+    // TODO(neelismail): Remove property key check once API supports age.
+    const entityCountPropertyValue =
+      entityCountStruct.definition?.[
+        property.key === "age" ? "year_of_birth" : property.key
+      ];
+
+    if (entityCountPropertyValue) {
+      let value =
+        entityCountPropertyValue.int64Val ??
+        entityCountPropertyValue.stringVal ??
+        entityCountPropertyValue.boolVal;
+
+      if (value) {
+        // TODO(neelismail): Remove age handling once the API supports them.
+        if (property.key === "age" && typeof value === "number") {
+          value = new Date().getFullYear() - value;
+        }
+
+        const definedValue = value;
+        if (property.buckets) {
+          property.buckets.forEach((range) => {
+            const min = range.min;
+            const max = range.max;
+            const displayName = range.displayName;
+            if (
+              (min && max && min <= definedValue && definedValue < max) ||
+              (min && !max && min <= definedValue) ||
+              (!min && max && definedValue < max)
+            ) {
+              propertyString = displayName;
+            }
+          });
+        } else {
+          propertyString = value.toString();
+        }
+      }
+    }
+    return propertyString;
+  };
 
   const fetchDemographicData = useCallback(async () => {
+    const groupByAttributes =
+      underlay.uiConfiguration.demographicChartConfigs.groupByAttributes;
+    const additionalSelectedAttributes = new Set<string>();
+    underlay.uiConfiguration.demographicChartConfigs.chartConfigs.forEach(
+      (config) => {
+        config.primaryProperties.forEach((property) => {
+          // TODO(neelismail): Remove guard for age property key when API provides age support
+          if (
+            !additionalSelectedAttributes.has(property.key) &&
+            (!groupByAttributes.includes(property.key) ||
+              (property.key === "age" &&
```

Comment:
Instead of handling "age" everywhere, treat it like everything else and replace it at the end.

Reply:
Done.